feature-push-notifications/src/main/java/io/novafoundation/nova/feature_push_notifications/data/domain/interactor/PushNotificationsInteractor.kt

```diff
@@ -50,4 +56,31 @@ class RealPushNotificationsInteractor(
     override fun isPushNotificationsEnabled(): Boolean {
         return pushSettingsProvider.isPushNotificationsEnabled()
     }
+
+    override fun isPushNotificationsAvailable(): Boolean {
+        return pushNotificationsService.isPushNotificationsAvailable()
+    }
+
+    override suspend fun onMetaAccountChanged(metaId: Long) {
+        val pushSettings = pushSettingsProvider.getPushSettings()
+        if (pushSettings.subscribedMetaAccounts.contains(metaId)) {
+            val isPushEnabled = pushSettingsProvider.isPushNotificationsEnabled()
+            updatePushSettings(isPushEnabled, pushSettings)
+        }
+    }
+
+    override suspend fun onMetaAccountRemoved(metaId: Long) {
+        val notificationsEnabled = pushSettingsProvider.isPushNotificationsEnabled()
+        val noAccounts = accountRepository.getActiveMetaAccountsQuantity() == 0
+        val pushSettings = pushSettingsProvider.getPushSettings()
+
+        if (notificationsEnabled && noAccounts) {
+            pushNotificationsService.updatePushSettings(false, null)
+        } else if (noAccounts) {
+            pushSettingsProvider.updateSettings(null)
```

Comment:
Why do we update notification settings to null while they should already be disabled? I think that's one of the places where logic error would become visible if we use sealed class for settings internal state

Reply:
It seems to me that we should reset the push settings when we delete the last meta account in the application, so that when the user creates a new meta account and enters the push settings, he will see the default settings

Reply:
So the initial logic was like that:
if notificationsEnabled && noAccounts - we do the first branch
else if noAccounts - we do the second branch

So if we are doing the second branch, the first one has failed => notificationsAreDisabled

I guess I've commented because I forgot that we're still storing notification settings even when notifications are disabled for easy re-enabling

Reply:
Here is what my thinking process looked like:
if `notificationsEnabled && noAccounts` - we do the first branch
else if `noAccounts` - we do the second branch

So if we are doing the second branch, the first one has failed => notificationsAreDisabled

I guess I've commented because I forgot that we're still storing notification settings even when notifications are disabled for easy re-enabling